src/instrumentation/instrumentation.erl

```diff
@@ -0,0 +1,66 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%%  Module that takes care of writing instrumentation to .csv file
+%%% @end
+%%%-------------------------------------------------------------------
+-module(instrumentation).
+-author("pawel").
+
+%% API
+-export([
+    create_ets/0,
+    start_link/0,
+    init_writer/0,
+    notify/1
+]).
+
+-define(MAX_BATCH_SIZE, 10000).
+
+-include("records.hrl").
+
+
+%%%---------------------------
+%% Exported functions
+%%%---------------------------
+
+create_ets() ->
+    ets:new(instrumentation, [named_table, public]),
+    ets:insert(instrumentation, {csv_writer, null}).
+
+start_link() ->
+    {ok, spawn_link(instrumentation, init_writer, [])}.
+
+notify(Notification) ->
+    ets:lookup_element(instrumentation, csv_writer, 2) ! Notification.
```

Comment:
Czy ty tutaj używasz etsa do znalezienia pida pewnego procesu? Raczej bym go po prostu zarejestrował https://erlang.org/doc/reference_manual/processes.html#registered-processes

Chyba że jest więcej niż jeden, ale patrząc na ten kod to nie wygląda